error-design/thinking-about-errors.md

```diff
@@ -47,4 +47,4 @@ Errors combine in all sorts of ways:
 
 ## Execution environment
 
-This one is a bit different to the others. It is important to consider how a program is executing when thinking about errors. We might want to treat errors differently when testing or debugging code, c.f., when the code is in production. We might also treat errors differently if our program is embedded in another (either as a library, a plugin, or some kind application-in-an-application-server scenario). Likewise, if our program is the OS, errors will be treated differently to if it is an application running on top of the OS. A program might also be inside some kind of VM or container with its own error handling facilities, or it might have some kind of supervisor or observer program.
+This one is a bit different to the others. It is important to consider how a program is executing when thinking about errors. We might want to treat errors differently when testing or debugging code, as opposed to, when the code is in production. We might also treat errors differently if our program is embedded in another (either as a library, a plugin, or some kind application-in-an-application-server scenario). Likewise, if our program is the OS, errors will be treated differently to if it is an application running on top of the OS. A program might also be inside some kind of VM or container with its own error handling facilities, or it might have some kind of supervisor or observer program.
```

Comment:
Hmm, this is a tricky one, maybe "versus"? Not sure how clear that is

Reply:
yeah 'versus' also works - the meaning is clear from context. In my view, the whole second part of the sentence is implied and can be omitted. 

```text
We might want to treat errors differently when testing or debugging code. We might also treat errors....
```

Reply:
Yeah, I think that reads well.